pkg/ddl/constraint.go

```diff
@@ -371,7 +371,7 @@ func (w *worker) verifyRemainRecordsForCheckConstraint(dbInfo *model.DBInfo, tab
 	// can let the check expression restored string as the filter in where clause directly.
 	// Prepare internal SQL to fetch data from physical table under this filter.
 	sql := fmt.Sprintf("select 1 from `%s`.`%s` where not %s limit 1", dbInfo.Name.L, tableInfo.Name.L, constr.ExprString)
-	ctx := kv.WithInternalSourceType(w.ctx, kv.InternalTxnDDL)
+	ctx := kv.WithInternalSourceType(w.workCtx, kv.InternalTxnDDL)
```

Comment:
What are the criteria for using `jobCtx.ctx` or `w.workerCtx`? I think using `jobCtx.ctx` is also reasonable here.

Reply:
Yes job context is better. I want the action has no related job to use worker's context. Here I forget to change the function signature to pass job context.